src/sage/schemes/elliptic_curves/weierstrass_morphism.py

```diff
@@ -973,6 +973,105 @@ def inseparable_degree(self):
         """
         return Integer(1)
 
+    def is_identity(self):
+        r"""
+        Check if this Weierstrass isomorphism is the identity.
+
+        EXAMPLES::
+
+            sage: from sage.schemes.elliptic_curves.weierstrass_morphism import WeierstrassIsomorphism
+            sage: p = 97
+            sage: Fp = GF(p)
+            sage: E = EllipticCurve(Fp, [1, 28])
+            sage: ws = WeierstrassIsomorphism(E, None, E)
+            sage: ws.is_identity()
+            False
+
+        ::
+
+            sage: from sage.schemes.elliptic_curves.weierstrass_morphism import WeierstrassIsomorphism
+            sage: p = 97
+            sage: Fp = GF(p)
+            sage: E = EllipticCurve(Fp, [1, 28])
+            sage: ws = WeierstrassIsomorphism(E, (1, 0, 0, 0), None)
```

Comment:
```suggestion
            sage: ws = E.isomorphism(1, 0, 0, 0)
```
Do you mind changing the constructors to use `.isomorphism` or `.isomorphism_to` for the rest as well?

Reply:
For this and also the above one: I think that since this is a "helper" class (i.e. people usually don't/shouldn't call it directly) it is better to put direct tests here. `.isomorphism` and `.isomorphism_to` are already tested and showcased elsewhere